### DISCUSSION TITLE
How to Specify the usage of POSIX version of MingW compiler? (Issues with std::mutex when cross-compiling)

### DISCUSSION THREAD

**Discussion by sierret** [microsoft/vcpkg#37885] [General]
I'd like to be able to specify that vcpkg uses the posix compatible variant of the MingW compiler. 

I am currently (trying to) cross-compile for Windows on a Linux system using MingW. However, I am encoutering errors such as 
`error: ‘mutex’ in namespace ‘std’ does not name a type`

I am pretty sure the issue comes down the Unix-Windows cross-compilation issues which I have actually already encountered and solved when build from source directly, i.e. outside of a package manager, by simply using the the posix compatible compiler version such as "i686-w64-mingw32-gcc-posix" instead of "i686-w64-mingw32-gcc".

However, I have no idea how to do this in VCPKG. Any help appreciated.

**Comment by dg0yt** [comment]
You want to use the posix threads variant of mingw gcc. That's probably also the one which has more testing here.

The vcpkg toolchain file for mingw looks for `<arch>-w64-mingw32-gcc` etc. with `find_program`.
One way to select binaries (if not on Windows) is to have a directory with symlinks with that name, pointing to the actual program, and add this directory to `PATH`.
On Ubuntu systems, the paths in `/usr/bin` are symlinks already: The actual tool is selected with `update-alternatives ...`.

**Reply by sierret** [reply]
Thanks so much! Been struggling with this for days. I feel like offering you my first-born. I was trying to configure cmake options in vcpkg then realised vcpkg was *generating* those options at "cmake runtime". Then tried playing around with vcpkg options but nothing seemed to work. I almost ended up having to manually build about a dozen libraries separately over on dirty nasty disgusting Windows and then manually include/link all those. 

I appreciate it but it all seems a bit roundabout tho. Is there no way to specify the posix threads variant in vcpkg settings?

I previously tried
`set(CMAKE_C_COMPILER i686-w64-mingw32-g++-posix)
set(CMAKE_CXX_COMPILER i686-w64-mingw32-gcc-posix)`

But that did jack shite.

**Reply by Neumann-A** [reply]
Learn this:
https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#useful-implementation-notes

Note: You need to have a toolchain/triplet for vcpkg to actually get stuff into vcpkg.
